src/_pytest/main.py

```diff
@@ -958,16 +964,41 @@ def collect(self) -> Iterator[nodes.Item | nodes.Collector]:
             self.trace.root.indent -= 1
 
     def genitems(self, node: nodes.Item | nodes.Collector) -> Iterator[nodes.Item]:
+        import inspect
+
+        from _pytest.python import Class
+        from _pytest.python import Function
+        from _pytest.python import Module
+
         self.trace("genitems", node)
         if isinstance(node, nodes.Item):
             node.ihook.pytest_itemcollected(item=node)
+            if self.config.getini("collect_imported_tests"):
```

Comment:
The logic for the variable is inverted: if `collect_imported_tests` is True, we are actually *excluding* imported tests instead of collecting them.

So the default for `collect_imported_tests` should be True for backward compatibility as discussed previously, and the logic here should be inverted:

```suggestion
            if not self.config.getini("collect_imported_tests"):
```